src/docker.rs

```diff
@@ -1351,26 +1368,17 @@ impl Docker {
     ) -> Result<Response<Incoming>, Error> {
         // This is where we determine to which transport we issue the request.
         let request = match *transport {
-            Transport::Http { ref client } => client.request(req),
+            #[cfg(feature = "http")]
+            Transport::Http { ref client } => client.request(req).map_err(Error::from).boxed(),
             #[cfg(feature = "ssl")]
-            Transport::Https { ref client } => client.request(req),
-            #[cfg(unix)]
-            Transport::Unix { ref client } => client.request(req),
-            #[cfg(windows)]
-            Transport::NamedPipe { ref client } => client.request(req),
+            Transport::Https { ref client } => client.request(req).map_err(Error::from).boxed(),
+            #[cfg(all(feature = "pipe", unix))]
+            Transport::Unix { ref client } => client.request(req).map_err(Error::from).boxed(),
+            #[cfg(all(feature = "pipe", windows))]
+            Transport::NamedPipe { ref client } => client.request(req).map_err(Error::from).boxed(),
             #[cfg(test)]
-            Transport::Mock { ref client } => client.request(req),
-            Transport::Custom { ref transport } => {
-                return match tokio::time::timeout(
-                    Duration::from_secs(timeout),
-                    transport.request(req),
-                )
-                .await
-                {
-                    Ok(v) => Ok(v?),
-                    Err(_) => Err(RequestTimeoutError),
-                };
```

Comment:
Just checking here... this is just removing a duplicate piece of code?

Reply:
Yes that's it, also it won't trigger a clippy warning when there is no feature enable,
this as the downside of needing boxed futures.

The change is in its own commit if we're not happy with the compromise.